include/zephyr/senss/senss.h

```diff
@@ -0,0 +1,384 @@
+ *
+ * @param buf The data buffer to receive sensor data sample.
+ *
+ * @param size The buffer size in bytes.
+ *
+ * @return 0 on success or negative error value on failure.
+ */
+int senss_read_sample(
+		int handle,
+		void *buf, int size);
+
+/**
+ * @brief Sensor data event receive callback.
+ *
+ * @param handle The sensor instance handle.
+ *
+ * @param buf The data buffer with sensor data.
+ *
+ * @param size The buffer size in bytes.
+ *
+ * @param param The user private parameter.
+ *
+ * @return 0 on success or negative error value on failure.
+ */
+typedef int (*senss_data_event_t)(
+		int handle,
+		void *buf, int size, void *param);
+
+/**
+ * @brief Register sensor data event receive callback.
+ *
+ * User can use this API to register a data event receive callback function.
+ *
+ * @param handle The sensor instance handle.
+ *
+ * @param callback The data event receive callback function.
+ *
+ * @param param The user private parameter.
+ *
+ * @return 0 on success or negative error value on failure.
+ */
+int senss_register_data_event_callback(
+		int handle,
+		senss_data_event_t callback, void *param);
+
+
+/**
+ * @brief Set report latency of a sensor instance.
+ *
+ * Report latency controls how long the sensor data events of the sensor instance will be buffered
+ * in the batching buffer before deliver to its client.
+ *
+ * For 0 latency value, it's means sensor data event will not buffered, and will report to its
+ * client immediately.
+ *
+ * The maximum sensor data events of this sensor instance is calculated as: latency /
+ * report interval.
+ *
+ * @param handle The sensor instance handle.
+ *
+ * @param latency The latency in micro seconds.
+ *
+ * @return 0 on success or negative error value on failure.
+ */
+int senss_set_report_latency(
+		int handle, uint64_t latency);
+
+/**
+ * @brief Get current report latency value of a sensor instance.
+ *
+ * @param handle The sensor instance handle.
+ *
+ * @param latency The value buffer to receive report latency in micro seconds.
+ *
+ * @return 0 on success or negative error value on failure.
+ */
+int senss_get_report_latency(
+		int handle, uint64_t *latency);
+
+/**
+ * @brief Flush batching buffer for a sensor instance.
+ *
+ * @param handle The sensor instance handle.
+ *
+ * If \p handle is -1, will flush all buffer.
+ *
+ * @return 0 on success or negative error value on failure.
+ */
+int senss_batching_flush(
+		int handle);
+
+/**
+ * @brief Get sensor information from sensor instance handle.
+ *
+ * @param handle The sensor instance hande.
+ *
+ * @return a const pointer to \ref senss_sensor_info on success or NULL on failure.
+ */
+const struct senss_sensor_info *senss_get_sensor_info(
+		int handle);
+
+/**
+ * @brief Get sensor instance's state
+ *
+ * @param handle The sensor instance hande.
+ *
+ * @param state Returned sensor state value
+ *
+ * @return 0 on success or negative error value on failure.
+ */
+int senss_get_sensor_state(
+		int handle,
```

Comment:
Use sensor_index, not handle, the client ID doesn't matter here

Reply:
sorry, as mentioned above, handle will be translated to connection, includes both reporter and client